core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/AsyncTransferProcessManager.java

```diff
@@ -278,6 +291,26 @@ private void run() {
         }
     }
 
+    private int processCommandQueue() {
+        var batchSize = 5;
+        var commands = commandQueue.dequeue(batchSize);
+        commands.forEach(command -> {
+            var commandResult = commandRunner.runCommand(command);
+            if (commandResult.failed()) {
+                //re-queue if possible
+                if (command.canRetry()) {
+                    monitor.warning(format("Could not process command [%s], will retry. error: %s", command.getClass(), commandResult.getFailureMessages()));
+                    commandQueue.enqueue(command);
+                } else {
+                    monitor.severe(format("Command [%s] has exceeded its retry limit, will discard now", command.getClass()));
+                }
+            } else {
+                monitor.debug(format("Successfully processed command [%s]", command.getClass()));
+            }
+        });
+        return 0;
```

Comment:
typo?

Reply:
not a typo, actually forgot to return the correct value :/